libs/sdk-flutter/android/src/main/kotlin/com/breez/breez_sdk/BreezSDKPlugin.kt

```diff
@@ -1,35 +1,82 @@
 import io.flutter.plugin.common.MethodChannel.Result
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
 
 /** BreezSDKPlugin */
-class BreezSDKPlugin: FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
-  private lateinit var channel : MethodChannel
-
-  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "breez_sdk")
-    channel.setMethodCallHandler(this)
-  }
-
-  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
+class BreezSDKPlugin : FlutterPlugin, MethodCallHandler, EventChannel.StreamHandler {
+    // Event has to be handled on main thread.
+    private var handler = Handler(Looper.getMainLooper())
+
+    /// The MethodChannel that will the communication between Flutter and native Android
+    ///
+    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
+    /// when the Flutter Engine is detached from the Activity
+    private lateinit var channel: MethodChannel
+    private var eventChannel: EventChannel? = null
+    private var eventSink: EventChannel.EventSink? = null
+    private var nodeLogStream: SdkLogListener? = null
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "breez_sdk")
+        channel.setMethodCallHandler(this)
+
+        eventChannel =
+            EventChannel(flutterPluginBinding.binaryMessenger, "breez_sdk_node_logs")
+        setNodeLogStream()
+        eventChannel?.setStreamHandler(this)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        if (call.method == "getPlatformVersion") {
+            result.success("Android ${android.os.Build.VERSION.RELEASE}")
+        } else {
+            result.notImplemented()
+        }
+    }
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
     }
-  }
 
-  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
-    channel.setMethodCallHandler(null)
-  }
+
+    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
+        eventSink = events
+    }
+
+    override fun onCancel(arguments: Any?) {
+        eventSink = null
+        eventChannel = null
+    }
+
+    fun setNodeLogStream(): SdkLogListener {
```

Comment:
This method is not static. It should be static and outside the plugin in order for the worker to be able to initialize the  singleton los stream.

Reply:
Yes, this function is not static, my thinking was that the worker has access to the [plugin instance](https://api.flutter.dev/javadoc/io/flutter/embedding/engine/plugins/PluginRegistry.html#get(java.lang.Class)) itself, so it's not being re-initialized, and it's always attached to Flutter engine beforehand.

I'll move the event channel outside and convert into a static function.

Reply:
Done.
[Convert setNodeLogStream to a static function](https://github.com/breez/breez-sdk/pull/727/commits/cb70ebe563ec58940cd21e92ebbb3d2c153b77db)